play-dl/YouTube/utils/extractor.ts

```diff
@@ -1,166 +1,187 @@
-            title : vid.title,
-            description : vid.shortDescription,
-            durationInSec : vid.lengthSeconds,
-            durationRaw : parseSeconds(vid.lengthSeconds),
-            uploadedDate : microformat.publishDate,
-            thumbnail : vid.thumbnail.thumbnails[vid.thumbnail.thumbnails.length - 1],
-            channel : {
-                name : vid.author,
-                id : vid.channelId,
-                url : `https://www.youtube.com/channel/${vid.channelId}`,
-                verified : Boolean(badge?.metadataBadgeRenderer?.style?.toLowerCase().includes('verified'))
-            },
-            views : vid.viewCount,
-            tags : vid.keywords,
-            averageRating : vid.averageRating,
-            live : vid.isLiveContent,
-            private : vid.isPrivate
-        }
-        if(!video_details.live) format.push(player_response.streamingData.formats[0])
-        format.push(...player_response.streamingData.adaptiveFormats)
-        let LiveStreamData = {
-            isLive : video_details.live,
-            dashManifestUrl : player_response.streamingData?.dashManifestUrl ?? null,
-            hlsManifestUrl : player_response.streamingData?.hlsManifestUrl ?? null
-        }
-        return {
-            LiveStreamData,
-            html5player,
-            format,
-            video_details
-        }
+export async function video_basic_info(url: string, cookie?: string) {
+    let video_id: string;
+    if (url.startsWith('https')) {
+        if (yt_validate(url) !== 'video') throw new Error('This is not a YouTube Watch URL');
+        video_id = extractID(url);
+    } else video_id = url;
+    const new_url = `https://www.youtube.com/watch?v=${video_id}`;
+    const body = await request(new_url, {
+        headers: cookie
+            ? {
+                  'cookie': cookie,
+                  'accept-language': 'en-US,en-IN;q=0.9,en;q=0.8,hi;q=0.7'
+              }
+            : { 'accept-language': 'en-US,en-IN;q=0.9,en;q=0.8,hi;q=0.7' }
+    });
+    const player_response = JSON.parse(body.split('var ytInitialPlayerResponse = ')[1].split('}};')[0] + '}}');
+    const initial_response = JSON.parse(body.split('var ytInitialData = ')[1].split('}};')[0] + '}}');
+    if (player_response.playabilityStatus.status !== 'OK')
+        throw new Error(
+            `While getting info from url\n${
+                player_response.playabilityStatus.errorScreen.playerErrorMessageRenderer?.reason.simpleText ??
+                player_response.playabilityStatus.errorScreen.playerKavRenderer?.reason.simpleText
+            }`
+        );
+    const badge =
+        initial_response.contents.twoColumnWatchNextResults.results.results.contents[1]?.videoSecondaryInfoRenderer
+            ?.owner?.videoOwnerRenderer?.badges &&
+        initial_response.contents.twoColumnWatchNextResults.results.results.contents[1]?.videoSecondaryInfoRenderer
+            ?.owner?.videoOwnerRenderer?.badges[0];
```

Comment:
I have some suggestion about badge code. coz we can use optional chaining to [](index operator) also.
So badge code can be like this.
```js
const badge =
        initial_response.contents.twoColumnWatchNextResults.results.results.contents[1]?.videoSecondaryInfoRenderer
            ?.owner?.videoOwnerRenderer?.badges?.[0];
```